src/main/scala/io/github/kiemlicz/shelm/ChartDownloader.scala

```diff
@@ -19,7 +19,26 @@ object ChartDownloader {
     * @param chartLocation Chart reference
     * @return directory containing Chart
     */
-  def download(chartLocation: ChartLocation, downloadDir: File, sbtLogger: Logger): File = {
+  def download(chartLocation: ChartLocation, downloadDir: File, cacheDir: Option[File], sbtLogger: Logger): File = {
```

Comment:
Thank you for this PR.

Couple of remarks:  
1. if the method can be private - please make it `private`  
2. I think that cache dir should be non-optional, check my other comment
3. You don't handle cache for all ChartLocations (e.g. `Remote` is not handled)
4. I don't think that Charts from different repositories should mix, they should have a different Cache key (introduce such type)
5. Don't use that many braces
6. [Please check this proposal of implementation (rough idea, not tested, contains unused method that should ultimately be deleted)]( https://gist.github.com/kiemlicz/b4b6c8ce1352bd2dbccf41e9d3fd3075)